nomad/config.go

```diff
@@ -444,6 +451,7 @@ func DefaultConfig() *Config {
 		ReconcileInterval:                60 * time.Second,
 		EvalGCInterval:                   5 * time.Minute,
 		EvalGCThreshold:                  1 * time.Hour,
+		BatchEvalGCThreshold:             168 * time.Hour,
```

Comment:
I think it makes sense to have this value "close to" the value of EvalGCThreshold for some definition of "close to." 24 hours is definitely closer, so this makes sense to me. 

That said, I harbor no strong opinion since it seems to me that users will very likely have wildly different approaches to this value (ranging from: "we don't care" to "we care strongly and think that value X is the only correct one -- the former won't care what the default is and the latter will disagree with any value we pick).

Changed to 24 hours by default.

> I don't have any strong opinion on the default value, but looking at it again whatever value we pick should be noted in the upgrade guide since we're going from infinity to some finite time 

Given that this is going to be a bug fix ported onto other versions, where should a note about this behavior change be placed? At each version in which this is included or just one version?

Let me know if you think I should include a note about this in the current pull request, or if it's something to be added prior to the release being cut

Reply:
I will open a follow-up PR to add the upgrade notice since this one will be backported.